config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/PrivateKeyEntityBuilder.java

```diff
@@ -69,12 +71,16 @@ private Entity buildPrivateKeyEntity(String alias, PrivateKey privateKey, Docume
     }
 
     private void buildAndAppendCertificateChainElement(PrivateKey privateKey, Element privateKeyElem, Document document) {
-        final KeyStore keyStore = keystoreHelper.loadKeyStore(privateKey);
-        final Certificate[] certificates = keystoreHelper.loadCertificatesForPrivateKey(privateKey, keyStore);
-
-        final Element[] certificatesElements = Stream.of(certificates)
-                .map(c -> createCertDataElementFromCert((X509Certificate) c, document)).toArray(Element[]::new);
-        privateKeyElem.appendChild(createElementWithChildren(document, CERTIFICATE_CHAIN, certificatesElements));
+        if (privateKey.getPrivateKeyFile() != null) {
+            final KeyStore keyStore = keystoreHelper.loadKeyStore(privateKey);
+            final Certificate[] certificates = keystoreHelper.loadCertificatesForPrivateKey(privateKey, keyStore);
+            final Element[] certificatesElements = Stream.of(certificates)
+                    .map(c -> createCertDataElementFromCert((X509Certificate) c, document)).toArray(Element[]::new);
+            privateKeyElem.appendChild(createElementWithChildren(document, CERTIFICATE_CHAIN, certificatesElements));
+        } else {
+            //add a dummy certificate
+            privateKeyElem.appendChild(createElementWithChildren(document, CERTIFICATE_CHAIN, CertificateUtils.createCertDataElementFromCert("", BigInteger.valueOf(0), "", "MIIBfTCCASegAwIBAgIJAPH69zKKw4ixMA0GCSqGSIb3DQEBBQUAMA8xDTALBgNVBAMTBHRlc3QwHhcNMTgxMDEzMDMyODI1WhcNMzgxMDA4MDMyODI1WjAPMQ0wCwYDVQQDEwR0ZXN0MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAIS+Vr8zPOBmSclkUtW/z0UXaMjhg7dix6IUZs+UoSiw/2GXfU2vc3renVAbn3AZaJEqnxgrcX4nldqt0WBIP4sCAwEAAaNmMGQwDgYDVR0PAQH/BAQDAgXgMBIGA1UdJQEB/wQIMAYGBFUdJQAwHQYDVR0OBBYEFN/aeDDEAB6MTxZhMhf/eJKnmaE5MB8GA1UdIwQYMBaAFN/aeDDEAB6MTxZhMhf/eJKnmaE5MA0GCSqGSIb3DQEBBQUAA0EAdolvh7bMX5ZMkM/yntJlBdzS8ukM/ULh8I11wKd6dDltyMuk9rOP0iEk1nsSFuFL0uQ4kIe12KyDwr8ns7VKvQ==", document)));
```

Comment:
isn't better to have this on a file rather than hardcoded? 
just a matter of design, no functional impact at all.

Reply:
Ya I don't like it either, but I also don't want to add additional logic to read this from a file. I'll make it a constant instead